src/abstractdataframe/reshape.jl

```diff
@@ -637,34 +641,53 @@ julia> permutedims(df2, 1, "different_name")
 """
 function Base.permutedims(df::AbstractDataFrame, src_namescol::ColumnIndex,
                           dest_namescol::Union{Symbol, AbstractString};
-                          makeunique::Bool=false)
+                          makeunique::Bool=false, strict::Bool=true)
 
     if src_namescol isa Integer
         1 <= src_namescol <= ncol(df) || throw(BoundsError(index(df), src_namescol))
     end
-    eltype(df[!, src_namescol]) <: SymbolOrString ||
-        throw(ArgumentError("src_namescol must have eltype `Symbol` or `<:AbstractString`"))
+    src_col_names = df[!, src_namescol]
+    local new_col_names
+    if eltype(src_col_names) <: SymbolOrString
+        new_col_names = src_col_names
+    elseif all(x -> x isa Symbol, src_col_names)
+        new_col_names = collect(Symbol, src_col_names)
+    elseif !strict
+        new_col_names = string.(src_col_names)
+    else
+        try
+            new_col_names = collect(AbstractString, src_col_names)
```

Comment:
Is there a reason to do `AbstractString` instead of just `String` or `Symbol` here? It feels a little weird to say it must be convert-able to `AbstractString`

Reply:
You mean that if `convert` to `AbstractString` exists for something then also there must be a `convert` to `String`? I think it should mostly hold, so I would be OK to change it here to `String`, as cases when you convert something to `AbstractString` but not to `String` are bad design.

@nalimilan - what do you think?

Reply:
I agree `String` sounds more logical. In practice this shouldn't really matter but it's simpler.